File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    ls = list(s)
    stk = []
    for i in ls:
        if i == "(":
            stk.append("(")
        else:
            if not len(stk):
                answer = False
                break
            elif stk[-1] == "(":
                stk.pop()
    if len(stk):
        answer = False
    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let stk = [];
    
    for (let i of s) {
        if (i === "("){
            stk.push("(");
        }
        else {
            if (stk.length === 0) {
                answer = false;
                break;
            }
            stk.pop();
        }
    }
    
    if (stk.length) {
        answer = false
    }

    return answer;
}